clang/test/CodeGenSYCL/int_header1.cpp

```diff
@@ -2,15 +2,15 @@
 // RUN: FileCheck -input-file=%t.h %s
 
 // CHECK:template <> struct KernelInfo<class KernelName> {
-// CHECK:template <> struct KernelInfo<::nm1::nm2::KernelName0> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName1> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3<::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName3<::nm1::KernelName1>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4<::nm1::nm2::KernelName0>> {
-// CHECK:template <> struct KernelInfo<::nm1::KernelName4<::nm1::KernelName1>> {
+// CHECK:template <> struct KernelInfo<class nm1::nm2::KernelName0> {
```

Comment:
I don't know if we can change this, @elizabethandrews @premanandrao , do you know?  We should possibly instead ensure that the printing policy does FQN instead?  Do either of you have an idea on how to do this?

Reply:
If you are asking about the global namespace scope resolution operator, I believe it is needed to ensure that 'nm1' is not looked up in the relative namespace scope.  @elizabethandrews had to do something similar; I think she found FQN didn't help with this and had to use a loop to insert the leading '::'.

Reply:
I think Elizabeth was talking about using a flag?

Reply:
I had to add a flag to printQualifiedName to emit the scope resolution operator at global scope. I believe Clang already has this support for getFullyQualifiedType.

You might need to add something similar to getCanonicalType.